src/platform_impl/windows/window.rs

```diff
@@ -912,60 +918,60 @@ unsafe fn register_window_class<T: 'static>(
     window_icon: &Option<Icon>,
     taskbar_icon: &Option<Icon>,
 ) -> Vec<u16> {
-    let class_name: Vec<_> = OsStr::new("Window Class")
+    let mut class_name: Vec<_> = OsStr::new("Window Class")
         .encode_wide()
         .chain(Some(0).into_iter())
         .collect();
```

Comment:
@kennykerr Do you think this is something we can replace with

```rust
    let class_name: windows::runtime::Param<'_, PWSTR> =
        windows::runtime::IntoParam::into_param("Window Class");
```
Or something more ergonomic? `IntoParam` works for function arguments but not for struct members.

Sidenote for windows-rs internals: I noticed you're allocating a new type/template parameter for every generic argument. Did you know you can also write `theParameter: impl IntoParam<...>` directly without having to come up with `ParamX`? That will save some characters and hopefully simplify codegen a little (but there are some annoying implementation details that we may or may not have already discussed before).

Reply:
Perhaps something like this: https://github.com/microsoft/windows-rs/issues/1307

> new type/template parameter for every generic argument

This is unfortunately necessary because Rust doesn't allow explicit generic arguments with the alternative syntax and this is required for some APIs that [cannot deduce the parameter type](https://github.com/microsoft/windows-rs/blob/master/crates/tests/win32/query/tests/test.rs).

Reply:
The macro way could work, or just a constructor on `P(W)STR` that the user can call themselves (for passing owned/boxed parameters in structs) in addition to `IntoParam` calling those constructors?
(After all I think that's how you'd implement the macro)

> This is unfortunately necessary because Rust doesn't allow explicit generic arguments with the alternative syntax and this is required for some APIs that [cannot deduce the parameter type](https://github.com/microsoft/windows-rs/blob/master/crates/tests/win32/query/tests/test.rs).

Right, we discussed that before and it's similar to using `Option<&dyn Something>` 
versus `Option<&impl Something>` where the latter needs an explicit type annotation if `None` is passed. I'm curious though if there are cases where this _can_ be done, but that's probably only making codegen needlessly more complicated.

Reply:
Also not possible due to disabled `std` feature